tests/unit/ui/components/TestShowMoreCard.js

```diff
@@ -1,105 +1,223 @@
     return shallowUntilTarget(
-      <ShowMoreCard i18n={fakeI18n()} {...props} />,
+      <ShowMoreCard {...props}>{children || 'some info'}</ShowMoreCard>,
       ShowMoreCardBase,
     );
-  };
+  }
 
-  it('reveals more text when clicking "show more" link', () => {
-    const root = render({ children: 'Hello I am description' });
+  it('reveals more text when clicking "read more" link', () => {
+    const { store } = dispatchClientMetadata();
+    const root = render({ store });
 
-    // We have to manually set the expanded flag to false because we don't have
-    // a clientHeight in the tests.
-    root.setState({ expanded: false });
+    // We are simulating the truncate method call.
+    root.instance().truncateToMaxHeight({ clientHeight: MAX_HEIGHT + 1 });
 
-    expect(root).not.toHaveClassName('ShowMoreCard--expanded');
+    applyUIStateChanges({ root, store });
 
-    // We have to shallow this prop because we pass a component to it.
-    expect(shallow(root.prop('footerLink')).html()).toContain(
-      '<span class="visually-hidden">Expand to</span> Read more',
-    );
+    expect(root).toHaveProp('footerLink');
+    const footerLink = root.prop('footerLink');
+    const cardLink = shallow(footerLink);
+    const moreLink = cardLink.find('.ShowMoreCard-expand-link');
 
-    // We cannot directly target the footer link, so we simulate the click here.
-    root.instance().onClick(createFakeEvent());
-    root.update();
+    moreLink.simulate('click', createFakeEvent());
 
-    expect(root).toHaveClassName('ShowMoreCard--expanded');
+    applyUIStateChanges({ root, store });
 
-    expect(root).toHaveProp('footerLink', null);
+    expect(root).toHaveClassName('ShowMoreCard--expanded');
   });
 
   it('is expanded by default', () => {
-    const root = render({ children: 'Hello I am description' });
+    const root = render();
 
     expect(root).toHaveClassName('ShowMoreCard--expanded');
   });
 
   it('truncates the contents if they are too long', () => {
-    const root = render({ children: 'Hello I am description' });
+    const { store } = dispatchClientMetadata();
+    const root = render({ store });
 
+    // We are simulating the truncate method call.
     root.instance().truncateToMaxHeight({ clientHeight: MAX_HEIGHT + 1 });
-    root.update();
+
+    applyUIStateChanges({ root, store });
 
     expect(root).not.toHaveClassName('ShowMoreCard--expanded');
   });
 
   it('renders className', () => {
     const className = 'test';
-    const root = render({
-      children: <p>Hi</p>,
-      className,
-    });
-
+    const root = render({ className });
     expect(root).toHaveClassName(className);
   });
 
   it('renders children', () => {
     const root = render({ children: 'Hello I am description' });
+    const contents = root.find('.ShowMoreCard-contents');
+    expect(contents).toHaveText('Hello I am description');
+  });
+
+  it('dispatches if the html children has changed', () => {
```

Comment:
I think this would make more sense as `dispatches if the children's html has changed`.